src/ray/gcs/gcs_server/task_info_handler_impl.cc

```diff
@@ -173,5 +173,12 @@ void DefaultTaskInfoHandler::HandleAttemptTaskReconstruction(
   }
 }
 
+void DefaultTaskInfoHandler::OnJobFinished(const JobID &job_id) {
+  RAY_CHECK_OK(gcs_table_storage_->TaskTable().DeleteByJobId(job_id, nullptr));
```

Comment:
This shouldn't be necessary. The new architecture and GCS-based actor management codepath never writes to the task table.

Reply:
ok. Heard that from @raulchen